soh/src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -145,17 +143,18 @@ void func_80A891C4(EnJs* this, PlayState* play) {
     if (Message_GetState(&play->msgCtx) == TEXT_STATE_CHOICE && Message_ShouldAdvance(play)) {
         switch (play->msgCtx.choiceIndex) {
             case 0: // yes
-                if (gSaveContext.rupees < 200) {
+                if (GameInteractor_Should(VB_CHECK_RANDO_PRICE_OF_CARPET_SALESMAN, gSaveContext.rupees < 200, this)) {
                     Message_ContinueTextbox(play, 0x6075);
                     func_80A89008(this);
                 } else {
-                    if (GameInteractor_Should(VB_GIVE_BOMBCHUS_FROM_CARPET_SALESMAN, true, this) || 
-                       (Actor_HasParent(&this->actor, play) || !GameInteractor_Should(VB_GIVE_ITEM_FROM_CARPET_SALESMAN, true, this))){
-                        Rupees_ChangeBy(-200);
-                        En_Js_SetupAction(this, func_80A89160);
-                    } else{
-                        Message_ContinueTextbox(play, 0x6073);
-                        func_80A89008(this);
+                    if (!GameInteractor_Should(VB_GIVE_ITEM_FROM_CARPET_SALESMAN, false, this)) {
+                        if (GameInteractor_Should(VB_GIVE_BOMBCHUS_FROM_CARPET_SALESMAN, true, this) || Actor_HasParent(&this->actor, play)){
+                            Rupees_ChangeBy(-200);
+                            En_Js_SetupAction(this, func_80A89160);
+                        } else{
+                            Message_ContinueTextbox(play, 0x6073);
+                            func_80A89008(this);
+                        }
                     }
```

Comment:
In order to avoid a 2 part knot of logic where Bombchu Bag and Merchant Shuffle are both trying to affect the same vanilla function, I made it so that if the merchant is selling a randomised check, it completely skips the vanilla handling and processes the sale entirely in the hook. This also made handling the flag and post-sale custom text (Now I can get back to being an arms dealer) timing conflict much easier as I could swap the text IDs without making another hook and another flag for that hook.